src/components/pages/UpdateFarmer/UpdateFarmer.js

```diff
@@ -11,6 +11,8 @@ import { toast } from 'react-toastify';
 const UpdateFarmer = ({ location, history, appStateShouldUpdate, user }) => {
   // Prevents errors when location state is empty
   const { farmer: farmerData } = location.state || {};
+  // state that keeps track of changed input fields
+  const [changes, setChanges] = useState({});
```

Comment:
You could probably not use setState() here, considering the component does not need to be rerendered based on the collected data.